src/entities/Basket/Basket.interface.ts

```diff
@@ -1,14 +1,24 @@
 import * as mongoose from "mongoose";
 
-type StatusType =
-  | "pending"
-  | "accepted"
-  | "rejected"
-  | "cancelled"
-  | "offered"
-  | "failedByRequestor"
-  | "failedByTarget"
-  | "success";
+type StatusType = 'pending'| 'accepted'|'rejected'|'cancelled'|'offered'|'failedByRequestor'|'failedByTarget'|'success';
+
+interface IStatusOrders extends Record<StatusType, number>  { }
+export const StatusOrder: IStatusOrders = {
+    'pending': 0,
```

Comment:
We decided to delete pending or offered  state

First step 0 - offered creates a basket = enters in phase pending/offered (one to choose from)
Second step 1 - One possibility for offerer, two for recipients: 
a) offerer withdraws the offer = status changed in cancelled (and you can't change it anymore! no step 3 in this case)
b) recipient rejects the offer = status changed in rejected (and you can't change it anymore! no step 3 in this case)
c) recipient accepts the offer = status changed in accepted (unblocks step 3)
Third step - two possibilites for offerer and two for recipient:
a) offerer doesn't show up - failedByRequestor (ends transaction)
b) recipient doesn't show up  - failedByTarget (ends transaction)
c) successOfferer - used to mark that for offerer the transaction was successful 
d) successRecipient - used to mark that for recipient the transaction was successful
If successOfferer comes after successRecipient or the other way round, automatically the third status should be set (instead of successRecipient or successOfferer)
a) success - the transaction ends

To sum up, we have the following ways to end the transaction with no coming back: cancelled, rejected, failedByRequestor, failedByRecipient, successOfferer/successRecipient (the second one sets status to succcess).
Therefore:
'pending': 0,
'accepted': 1
'successRecipient', 'successOfferer': 2
'rejected', 'cancelled', 'failedByRequestor, 'failedByTarget', 'success': 3 (all exclude further transactions)

However, I'm not sure if it's a good idea to use enums here, because this schema is still not complete and takes into consideration only going back, not the order in which the requests should be sent - for example you shoudn't be able to pass directly from pending to failedBy requestor etc. 

What is more, we should also check the status of the person who sends the request - for offerer it should not be possible to reject, accept, set failedByRequestor, successRecipient ect.

What can we do? I think about the classical if statements (it will be hurtful but and personally don't see another option. We can leave it imperfect and only improve you enums (add succesRecipient, successRequestor and automatic changing into success after the second person marks transaction as successful, delete one from pending/offered).

Reply:
Good point about blocking certain statuses for requestor/recipient. I did not block failedByRequestor/failedByTarget because there is no harm if someone failed and blames himself. 
I added blockers for skipping statuses all failures are permanent and marked by -1
One thing that I skipped is that failureBy status is allowed  before accepting basket by another user since we don't be able to prevent people from false accusation anyway it rather does not matter